### PR TITLE
Add test to check netsh active response alert when firewall is disabled in Windows

### PR DESCRIPTION
**Pull request by fedepacher**
|Related issue|
|-------------|
|     https://github.com/wazuh/wazuh-qa/issues/3296       |

## Description

This PR aims to add tests to check if `netsh active-response` works correctly generating an alert when the firewall is disabled, It is written in the active response log in order to be able to generate the alert.

### Added

- get_alert_json.yaml
- search_alert.yaml
- truncate_alert_json.yaml
- get_log_file.yaml
- restart_wazuh.yaml
- write_wazuh_config.yaml
- write_wazuh_internal_options.yaml
- configuration.yaml
- generate_events.yaml
- teardown.yaml
- cases_firewall_alerts.yaml
- cases_firewall_status.yaml
- test_firewall_status.py
- test_firewall_alerts.py
- conftest.py

## Testing performed

| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @fedepacher (Developer)  |           | [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/10456864/r1.zip) |⚫⚫⚫  |         |         | Nothing to highlight |
|  @jmv74211  (Reviewer)   |           | ⚫⚫⚫ | ⚫⚫⚫  |        |         | Nothing to highlight |

### CODE REVIEW

File: tests/system/test_active_response/test_netsh_windows_command/test_firewall_alerts/data/playbooks/configuration.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+- name: Configure Windows agent environment
+  hosts: windows-agent
+  tasks:
+
+    - name: Create temp folder
+      win_file:
+        path: C:\temp\
+        state: directory
+
+    - name: Copy ossec.conf
```
Comment (jmv74211) [inline tests/system/test_active_response/test_netsh_windows_command/test_firewall_alerts/data/playbooks/configuration.yaml:36]:
```suggestion
    - name: Make a backup of ossec.conf
```

File: tests/system/test_active_response/test_netsh_windows_command/test_firewall_alerts/data/playbooks/generate_events.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+- name: Clean alerts file
+  hosts: manager
+  tasks:
+
+    - name: Truncate alert.json
+      include_role:
+        name: manage_alerts
+        tasks_from: truncate_alert_json.yaml
```
Comment (jmv74211) [inline tests/system/test_active_response/test_netsh_windows_command/test_firewall_alerts/data/playbooks/generate_events.yaml:8]: Move this to another playbook

File: tests/system/test_active_response/test_netsh_windows_command/test_firewall_alerts/data/test_cases/cases_firewall_alerts.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+- name: firewall_disabled
+  description: Check that the firewall is disabled.
```
Comment (jmv74211) [inline tests/system/test_active_response/test_netsh_windows_command/test_firewall_alerts/data/test_cases/cases_firewall_alerts.yaml:2]: Update this

File: tests/system/test_active_response/test_netsh_windows_command/test_firewall_alerts/test_firewall_alerts.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                    error_message=f"Could not find the event '{expected_log}' in alerts.json file")
+
+
+@pytest.mark.parametrize('metadata', configuration_metadata, ids=cases_ids)
+def test_firewall_alerts(configure_environment, metadata, generate_events):
+    '''
+    description: Check that an alert is generated when the firewall is disabled.
+
+    test_phases:
+        - Set a custom Wazuh configuration.
```
Comment (jmv74211) [inline tests/system/test_active_response/test_netsh_windows_command/test_firewall_alerts/test_firewall_alerts.py:40]: Complete this according to the documentation structure

File: tests/system/test_active_response/test_netsh_windows_command/test_firewall_alerts/test_firewall_alerts.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        - The `configuration.yaml` file provides the module configuration for this test.
+        - The `generate_events.yaml` file provides the function to copy the log file to a temporary folder, provides
+          the playbook to insert the log into the test files and, enable/disable the firewall.
+    '''
+    status = metadata['extra_vars']['firewall_status']
+    for expected_log in metadata['extra_vars']['firewall_status_logs']:
+        if 'disabled' in status:
+            check_event(expected_log)
+        else:
+            with pytest.raises(TimeoutError):
```
Comment (jmv74211) [inline tests/system/test_active_response/test_netsh_windows_command/test_firewall_alerts/test_firewall_alerts.py:73]: Add a comment explaining what is doing /checking

File: tests/system/test_active_response/test_netsh_windows_command/test_firewall_alerts/test_firewall_alerts.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+
+@pytest.mark.parametrize('metadata', configuration_metadata, ids=cases_ids)
+def test_firewall_alerts(configure_environment, metadata, generate_events):
+    '''
+    description: Check that an alert is generated when the firewall is disabled.
+
+    test_phases:
+        - Set a custom Wazuh configuration.
+        - Generates RDP attacks
```
Comment (jmv74211) [inline tests/system/test_active_response/test_netsh_windows_command/test_firewall_alerts/test_firewall_alerts.py:41]: Update this

File: tests/system/test_active_response/test_netsh_windows_command/test_firewall_status/data/playbooks/configuration.yaml [hunk 7]
```diff
@@ -0,0 +1,96 @@
+- name: Configure manager environment
+  hosts: manager
+  become: true
+  tasks:
+
+    - name: Update packages list
+      ansible.builtin.apt:
```
Comment (jmv74211) [inline tests/system/test_active_response/test_netsh_windows_command/test_firewall_status/data/playbooks/configuration.yaml:7]: ` apt`? What happens if we have a RPM manager?

File: tests/system/test_active_response/test_netsh_windows_command/test_firewall_status/data/playbooks/generate_events.yaml [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+    - name: Attempt a RDP brute force attack
+      shell: hydra -l {{ item }} -p invalid_password rdp://{{ hostvars['windows-agent']['ansible_host'] }}
+      loop:
+        - test_user
+
+- name: Copy log file to a tmp folder
+  hosts: windows-agent
+  tasks:
+
```
Comment (jmv74211) [inline tests/system/test_active_response/test_netsh_windows_command/test_firewall_status/data/playbooks/generate_events.yaml:25]: We don't need any waiting time?

File: tests/system/test_active_response/test_netsh_windows_command/test_firewall_status/test_firewall_status.py [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+          provides the playbook to generate the RDP attack.
+    '''
+    status = metadata['extra_vars']['firewall_status']
+    for expected_log in metadata['extra_vars']['firewall_status_logs']:
+        if 'disabled' in status:
+            check_event(expected_log)
+        else:
+            with pytest.raises(TimeoutError):
+                check_event(expected_log)
+                raise AttributeError(f'Unexpected log {expected_log}')
```
Comment (jmv74211) [inline tests/system/test_active_response/test_netsh_windows_command/test_firewall_status/test_firewall_status.py:77]: The alert `x` was generated unexpectedly.

**Review by jmv74211 — CHANGES_REQUESTED** [review]

File: tests/system/test_active_response/test_netsh_windows_command/test_firewall_status/test_firewall_status.py (hunk 9 above)
Comment (fedepacher) [reply tests/system/test_active_response/test_netsh_windows_command/test_firewall_status/test_firewall_status.py]: Done [here](https://github.com/wazuh/wazuh-qa/pull/3787/commits/e4ded61433bf9f440dd1336be139e4ecb01581ec)

File: tests/system/test_active_response/test_netsh_windows_command/test_firewall_status/data/playbooks/generate_events.yaml (hunk 8 above)
Comment (fedepacher) [reply tests/system/test_active_response/test_netsh_windows_command/test_firewall_status/data/playbooks/generate_events.yaml:32]: Done [here](https://github.com/wazuh/wazuh-qa/pull/3787/commits/e4ded61433bf9f440dd1336be139e4ecb01581ec)

File: tests/system/test_active_response/test_netsh_windows_command/test_firewall_status/data/playbooks/configuration.yaml (hunk 7 above)
Comment (fedepacher) [reply tests/system/test_active_response/test_netsh_windows_command/test_firewall_status/data/playbooks/configuration.yaml:7]: Done [here](https://github.com/wazuh/wazuh-qa/pull/3787/commits/e4ded61433bf9f440dd1336be139e4ecb01581ec)

File: tests/system/test_active_response/test_netsh_windows_command/test_firewall_alerts/test_firewall_alerts.py (hunk 6 above)
Comment (fedepacher) [reply tests/system/test_active_response/test_netsh_windows_command/test_firewall_alerts/test_firewall_alerts.py]: Done [here](https://github.com/wazuh/wazuh-qa/pull/3787/commits/e4ded61433bf9f440dd1336be139e4ecb01581ec)

File: tests/system/test_active_response/test_netsh_windows_command/test_firewall_alerts/test_firewall_alerts.py (hunk 4 above)
Comment (fedepacher) [reply tests/system/test_active_response/test_netsh_windows_command/test_firewall_alerts/test_firewall_alerts.py]: Done [here](https://github.com/wazuh/wazuh-qa/pull/3787/commits/e4ded61433bf9f440dd1336be139e4ecb01581ec)

File: tests/system/test_active_response/test_netsh_windows_command/test_firewall_alerts/data/test_cases/cases_firewall_alerts.yaml (hunk 3 above)
Comment (fedepacher) [reply tests/system/test_active_response/test_netsh_windows_command/test_firewall_alerts/data/test_cases/cases_firewall_alerts.yaml]: Done [here](https://github.com/wazuh/wazuh-qa/pull/3787/commits/e4ded61433bf9f440dd1336be139e4ecb01581ec)

File: tests/system/test_active_response/test_netsh_windows_command/test_firewall_alerts/data/playbooks/generate_events.yaml (hunk 2 above)
Comment (fedepacher) [reply tests/system/test_active_response/test_netsh_windows_command/test_firewall_alerts/data/playbooks/generate_events.yaml]: Done [here](https://github.com/wazuh/wazuh-qa/pull/3787/commits/e4ded61433bf9f440dd1336be139e4ecb01581ec)

File: tests/system/test_active_response/test_netsh_windows_command/test_firewall_alerts/data/playbooks/configuration.yaml (hunk 1 above)
Comment (fedepacher) [reply tests/system/test_active_response/test_netsh_windows_command/test_firewall_alerts/data/playbooks/configuration.yaml]: Done [here](https://github.com/wazuh/wazuh-qa/pull/3787/commits/e4ded61433bf9f440dd1336be139e4ecb01581ec)

File: CHANGELOG.md [hunk 10]
```diff
@@ -9,6 +9,7 @@ Release report: TBD
 
 ### Added
 
+- Add test to check if active-response netsh generates alerts when firewall is disabled. ([3787](https://github.com/wazuh/wazuh-qa/pull/3787)) \- (Framework + Tests)
```
[the remark this answers is not in the document]
Comment (fedepacher) [reply CHANGELOG.md]: Done [here](https://github.com/wazuh/wazuh-qa/pull/3787/commits/e4ded61433bf9f440dd1336be139e4ecb01581ec)

File: CHANGELOG.md [hunk 11]
```diff
@@ -21,6 +22,7 @@ Release report: TBD
 
 ### Fixed
 
+- Fix a regex error in the FIM integration tests ([#3061](https://github.com/wazuh/wazuh-qa/issues/3061)) \- (Framework + Tests)
```
[the remark this answers is not in the document]
Comment (fedepacher) [reply CHANGELOG.md:26]: Done [here](https://github.com/wazuh/wazuh-qa/pull/3787/commits/e4ded61433bf9f440dd1336be139e4ecb01581ec)

File: tests/system/test_active_response/test_netsh_windows_command/test_firewall_alerts/test_firewall_alerts.py [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+            - Load Wazuh light configuration.
+            - Apply ossec.conf configuration changes according to the configuration template and use case.
+            - Truncate wazuh logs.
+            - Restart wazuh-manager service to apply configuration changes.
+            - Change status to the Windows firewall.
+            - Insert log into a monitorized file
+        - test:
+            - Check in the alerts.json that an alerts of firewall disabled are generated when the firewall is disabled.
+            - Check in the alerts.json that no alerts of firewall enabled are generated when the firewall is enabled.
+        - tierdown:
```
Comment (jmv74211) [inline tests/system/test_active_response/test_netsh_windows_command/test_firewall_alerts/test_firewall_alerts.py:41]:
```suggestion
        - teardown:
```

File: tests/system/test_active_response/test_netsh_windows_command/test_firewall_alerts/test_firewall_alerts.py [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+
+
+# Test cases data
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+test_cases_file_path = os.path.join(test_data_path, 'test_cases', 'cases_firewall_alerts.yaml')
+alerts_json = os.path.join(gettempdir(), 'alerts.json')
+
+# Playbooks
+configuration_playbooks = ['configuration.yaml']
+# truncate_files_playbooks = ['truncate_files.yaml']
```
Comment (jmv74211) [inline tests/system/test_active_response/test_netsh_windows_command/test_firewall_alerts/test_firewall_alerts.py:17]: Remove this if you are not using it.

File: tests/system/test_active_response/test_netsh_windows_command/test_firewall_status/test_firewall_status.py [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+            - Truncate wazuh logs.
+            - Restart wazuh-manager service to apply configuration changes.
+            - Change status to the Windows firewall.
+            - Generate an RDP attack to the Windows agent
+        - test:
+            - Check in the active-responses.log that a log for firewall disabled is generated when the firewall is
+              disabled.
+            - Check in the active-responses.log that no log for firewall enabled is generated when the firewall is
+              enabled.
+        - tierdown:
```
Comment (jmv74211) [inline tests/system/test_active_response/test_netsh_windows_command/test_firewall_status/test_firewall_status.py:44]:
```suggestion
        - teardown:
```

**Review by jmv74211 — CHANGES_REQUESTED** [review]

**Review by jmv74211 — APPROVED** [review]

File: tests/system/test_active_response/test_netsh_windows_command/test_firewall_status/test_firewall_status.py (hunk 14 above)
Comment (fedepacher) [reply tests/system/test_active_response/test_netsh_windows_command/test_firewall_status/test_firewall_status.py]: Done [here](https://github.com/wazuh/wazuh-qa/pull/3787/commits/467db3146f4d65a5fc23f1e18584af623236e640)

File: tests/system/test_active_response/test_netsh_windows_command/test_firewall_alerts/test_firewall_alerts.py (hunk 12 above)
Comment (fedepacher) [reply tests/system/test_active_response/test_netsh_windows_command/test_firewall_alerts/test_firewall_alerts.py]: Done [here](https://github.com/wazuh/wazuh-qa/pull/3787/commits/467db3146f4d65a5fc23f1e18584af623236e640)

**Pull request merged**